examples/ecdsa-psbt.rs

```diff
@@ -0,0 +1,629 @@
+/// An example of an offline signer i.e., a cold-storage device.
+struct ColdStorage {
+    /// The master extended private key.
+    master_xpriv: ExtendedPrivKey,
+    /// The master extended public key.
+    master_xpub: ExtendedPubKey,
+}
+
+/// The data exported from an offline wallet to enable creation of a watch-only online wallet.
+/// (wallet, fingerprint, account_0_xpub, input_utxo_xpub)
+type ExportData = (ColdStorage, Fingerprint, ExtendedPubKey, ExtendedPubKey);
+
+impl ColdStorage {
+    /// Constructs a new `ColdStorage` signer.
+    ///
+    /// # Returns
+    ///
+    /// The newly created signer along with the data needed to configure a watch-only wallet.
+    fn new<C: Signing>(secp: &Secp256k1<C>, xpriv: &str)-> Result<ExportData> {
+        let master_xpriv = ExtendedPrivKey::from_str(xpriv)?;
+        let master_xpub = ExtendedPubKey::from_priv(secp, &master_xpriv);
+
+        // Hardened children require secret data to derive.
+
+        let path = "m/84h/0h/0h".into_derivation_path()?;
+        let account_0_xpriv = master_xpriv.derive_priv(secp, &path)?;
+        let account_0_xpub = ExtendedPubKey::from_priv(secp, &account_0_xpriv);
+
+        let path = INPUT_UTXO_DERIVATION_PATH.into_derivation_path()?;
+        let input_xpriv = master_xpriv.derive_priv(secp, &path)?;
+        let input_xpub = ExtendedPubKey::from_priv(secp, &input_xpriv);
+
+        let wallet = ColdStorage {
+            master_xpriv,
+            master_xpub,
+        };
+        let fingerprint = wallet.master_fingerprint();
+
+        Ok((wallet, fingerprint, account_0_xpub, input_xpub))
+    }
+
+    /// Returns the fingerprint for the master extended public key.
+    fn master_fingerprint(&self) -> Fingerprint {
+        self.master_xpub.fingerprint()
+    }
+
+    /// Signs `psbt` with this signer.
+    fn sign_psbt<C: Signing>(&self, secp: &Secp256k1<C>, mut psbt: Psbt) -> Result<Psbt> {
+        let sk = self.private_key_to_sign(secp, &psbt.inputs[0])?;
+        psbt_sign::sign(&mut psbt, &sk, 0, secp)?;
+
+        Ok(psbt)
+    }
+
+    /// Returns the private key required to sign `input` if we have it.
+    fn private_key_to_sign<C: Signing>(&self, secp: &Secp256k1<C>, input: &Input) -> Result<PrivateKey> {
+        match input.bip32_derivation.iter().next() {
+            Some((pk, (fingerprint, path))) => {
+                if *fingerprint != self.master_fingerprint() {
+                    return Err(Error::WrongFingerprint);
+                }
+
+                let sk = self.master_xpriv.derive_priv(secp, &path)?.to_priv();
+                if *pk != sk.public_key(secp).inner {
+                    return Err(Error::WrongPubkey);
+                }
+
+                Ok(sk)
+
+            },
+            None => Err(Error::MissingBip32Derivation),
+        }
+    }
+}
+
+/// An example of an watch-only online wallet.
+struct WatchOnly {
```

Comment:
I would not call it `WatchOnly` because watch wallets do not create any transactions or assist in creating those. They just watch the blockchain for funds. This is more of an example of a software wallet that actively creates/finalizes transactions with an offline HW signer.

Reply:
After some reading, I think WatchOnly is the correct term here